selfhost/lexer.jakt

```diff
@@ -774,7 +774,14 @@ struct Lexer {
         }
         loop {
             let ch = .peek()
-            if ch == b' ' or ch == b'\t' or ch == b'\r' {
+            if ch == 0 {
+                // ch == 0 means that .peek() has detected the end of the input
+                // stream and reported with a Token:Eof using the same construct
+                // for the token as above. This includes incrementing the index, 
+                // even if it is already past the end of the input array.
```

Comment:
Instead of relying on `.peek()` to return `0`, let's explicitly check for `.eof()` here instead.

Reply:
> Instead of relying on `.peek()` to return `0`, let's explicitly check for `.eof()` here instead.

Sure - also makes it express the intention more clearly. However, that would effectively put four checks for EOF in a row:
- One at the beginning of the function by checking `.index == .input.size()`
- One right after to check if the index has been incremented further
- One at the top of the loop
- One inside eof() (which then returns 0 if at EOF)

To avoid adding more tests for specific cases, I suggest to put it all in the loop, which has the purpose of consuming white space and bail out if EOF is encountered. Left a FIXME for the `if` block that returns `None`. Once that FIXME is handled, this will leave only a single test for EOF in the function.

I believe that this makes for cleaner code, but most happy to hear if that is not the case.